File: src/base_files/nodejs/core/errors/failures.ts
class Failure {
    code: number;
    message: string;

    constructor(code: number, message: string) {
        this.code = code;
        this.message = message;
    }
}
class ServiceFailure extends Failure {
    code: number;
    message: string;

    constructor(code: number, message: string) {
        super(code, message);

        this.code = code;
        this.message = message;
    }
}

export { Failure, ServiceFailure };

File: src/base_files/dart/core/errors/failures.dart
import 'package:equatable/equatable.dart';

abstract class Failure extends Equatable {
  final int code;
  final String message;

  const Failure({required this.code, required this.message});

  @override
  List<Object> get props => [code, message];
}

class ServerFailure extends Failure {
  const ServerFailure({required super.code, required super.message});
}

class CacheFailure extends Failure {
  const CacheFailure({required super.code, required super.message});
}
